.github/scripts/sign.ps1

```diff
@@ -0,0 +1,135 @@
+    "SignBatches": [
+        {
+            "SourceLocationType": "UNC",
+            "SourceRootDirectory": "$workspace\\unsigned",
+            "DestinationLocationType": "UNC",
+            "DestinationRootDirectory": "$workspace\\signed",
+            "SignRequestFiles": [
+                {
+                    "SourceLocation": "$fileName",
+                    "DestinationLocation": "$fileName"
+                }
+            ],
+            "SigningInfo": {
+                "Operations": [
+                    {
+                        "KeyCode": "CP-401405",
+                        "OperationCode": "NuGetSign",
+                        "ToolName": "sign",
+                        "ToolVersion": "1.0"
+                    },
+                    {
+                        "KeyCode": "CP-401405",
+                        "OperationCode": "NuGetVerify",
+                        "ToolName": "sign",
+                        "ToolVersion": "1.0"
+                    }
+                ]
+            }
+        }
+    ]
+}
+"@
+Out-File -FilePath .\auth.json -InputObject $authJson
+Out-File -FilePath .\input.json -InputObject $inputJson
+Write-Host 'Done.'
+try {
+    Write-Host 'Downloading ESRP Client.'
+    az storage blob download --auth-mode login --subscription  $subscriptionId --account-name $storage_name -c $container_name -n microsoft.esrpclient.1.2.76.zip -f esrp.zip
+    if (Test-Path 'esrp.zip') {
+        Write-Host 'Done.'
+    } else {
+        throw 'Download did not complete successfully. This is likely due to an access issue.'
+    }
+    
+    Write-Host 'Unzipping ESRP Client.'
+    Expand-Archive -Path 'esrp.zip' -DestinationPath './esrp' -Force
+    Write-Host 'Done.'
+    Write-Host 'Downloading & Installing Certifictes.'
+    Remove-Item cert.pfx -ErrorAction SilentlyContinue
+    az keyvault secret download --subscription $subscriptionId --vault-name $vault_name --name $aad_cert -f cert.pfx
+    certutil -silent -f -importpfx cert.pfx
+    Remove-Item cert.pfx
+    az keyvault secret download --subscription $subscriptionId --vault-name $vault_name --name $sign_cert -f cert.pfx
+    certutil -silent -f -importpfx cert.pfx
+    Remove-Item cert.pfx
+    Write-Host 'Done.'
+    Write-Host 'Executing ESRP Client.'
+    ./esrp/tools/EsrpClient.exe sign -a ./auth.json -p ./esrp/tools/Policy.json -c ./esrp/tools/Config.json -i ./input.json -o ./Output.json -l Verbose -f STDOUT
+    Write-Host 'Done. Signing Complete.'
+    Write-Host 'Verifying signatures with NuGet.'
+    $result = nuget verify -Signatures signed/$fileName -CertificateFingerprint $signing_cert_fingerprint
+    $anyMatches = $result | Where-Object { $_ -match 'Package signature validation failed.'}
+    if ([string]::IsNullOrWhiteSpace($anyMatches)) {
+        Write-Host 'Done. Signatures verified.'
+        Write-Host 'Package ready for upload.'
+    } else {
+        throw 'Package signature validation failed.'
+    }
+} catch {
+    throw
```

Comment:
In what cases would this trigger?

Reply:
Currently, it would throw if downloading of the ESRP Client failed, if something in the esrp.exe threw, or if the nuget verify flow failed.